lib/ddtrace/tracer.rb

```diff
@@ -230,9 +230,10 @@ def start_span(name, options = {})
         # root span
         @sampler.sample!(span)
         span.set_tag('system.pid', Process.pid)
-        if ctx && ctx.trace_id && ctx.span_id
-          span.trace_id = ctx.trace_id
-          span.parent_id = ctx.span_id
+
+        if ctx
+          span.trace_id = ctx.trace_id unless ctx.trace_id.nil?
+          span.parent_id = ctx.span_id unless ctx.span_id.nil?
```

Comment:
We might want this as.

```ruby
span.trace_id = ctx.trace_id unless ctx.trace_id.nil?
span.parent_id = ctx.span_id unless ctx.trace_id.nil? && ctx.span_id.nil?
```

Wondering if there are any potential issues with us inheriting the parent id but not the trace id?

or maybe we need:

```ruby
if ctx && ctx.trace_id
  span.trace_id = ctx.trace_id
  span.parent_id = ctx.span_id unless ctx.span_id.nil?
end
```